[docs][ybm] Add Azure support (#17729)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-cloud/cloud-basics/cloud-vpcs/managed-endpoint-azure.md

      Azure
    </a>
  </li>

</ul>

To use Azure Private Link, you need the following:

- An Azure user account with an active subscription.
- An Azure service that supports private endpoints.
- The [subscription ID](https://learn.microsoft.com/en-us/azure/azure-portal/get-subscription-tenant-id) of the service to which to grant access to the endpoint.

Make sure that default security group in your application VPC allows internal connectivity. Otherwise, your application may not be able to reach the endpoint.

## Create a PSE for Azure Private Link using ybm CLI

To use Private Link to connect your cluster to a VPC that hosts your application, first create a PSE on your cluster, then create an endpoint in Azure.

### Create a PSE in YugabyteDB Managed

To create a PSE, do the following:

1. Enter the following command:

    ```sh
    ybm cluster network endpoint create \
      --cluster-name <yugabytedb_cluster> \
      --region <cluster_region> \
      --accessibility-type PRIVATE_SERVICE_ENDPOINT \
      --subscription-id <subscription_ids>
    ```

    Replace values as follows:

    - `yugabytedb_cluster` - name of your cluster.

>     - `cluster_region` - cluster region where you want to place the PSE. Must match one of the regions where your cluster is deployed (for example, `us-west-2`), as well as the region where your application is deployed.

Reviewer: ```suggestion
    - `cluster_region` - cluster region where you want to place the PSE. Must match one of the regions where your cluster is deployed (for example, `us-west-2`), and preferably match the region where your application is deployed.
```

    ```sh
    az network private-endpoint show \
    --name <private_endpoint_name> \
    --resource-group <resource_group_name>
    ```

    This command returns the following values that you use to link the private DNS zone to the VNET:

    - ipv4 address of the private endpoint (`private_endpoint_ipv4_address` in the following commands)
    - VNet of the private endpoint (`private_endpoint_vnet` in the following commands)

1. To link the private DNS zone to the VNet containing the private endpoint, enter the following command:

    ```sh
    az network private-dns link vnet create 
        --name <private_dns_zone_name>
        --registration-enabled true
        --resource-group <resource_group_name>
        --virtual-network <private_endpoint_vnet>
        --zone-name <cluster_id>.azure.ybdb.io
        --tags yugabyte
    ```

    Replace values as follows:

    - `private_dns_zone_name` - provide a name for the private DNS zone.
    - `resource_group_name` - the resource group in which the private endpoint was created.
    - `private_endpoint_vnet` - the VNet of the private endpoint.

>     - `cluster_id` - the cluster ID of the cluster with the PSE; the cluster ID is displayed on the cluster **Settings** tab in YBM.

Reviewer: same comment as above -
"@ddhodge we will auto-generate this entire value as we know which cluster user is creating private endpoint for. I left it as <cluster_id> because this is based on the cluster. Can you please message this accordingly?"

Make sure that default security group in your application Azure Virtual Network (VNet) allows internal connectivity. Otherwise, your application may not be able to reach the endpoint.

## Create a PSE for Azure Private Link using ybm CLI

To use Private Link to connect your cluster to an VNet that hosts your application, first create a PSE on your cluster, then create an endpoint in Azure.

### Create a PSE in YugabyteDB Managed

To create a PSE, do the following:

1. Enter the following command:

    ```sh
    ybm cluster network endpoint create \
      --cluster-name <yugabytedb_cluster> \
      --region <cluster_region> \
      --accessibility-type PRIVATE_SERVICE_ENDPOINT \
      --security-principals <subscription_ids>
    ```

    Replace values as follows:

    - `yugabytedb_cluster` - name of your cluster.
    - `cluster_region` - cluster region where you want to place the PSE. Must match one of the regions where your cluster is deployed (for example, `us-west-2`), and preferably match the region where your application is deployed.

>     - `subscription_ids` - comma-separated list of the subscription IDs of services that you want to grant access.

Reviewer 2: subscription IDs "of Azure subscriptions"

- An Azure user account with an active subscription.
- The subscription ID of the service to which to grant access to the cluster endpoint.

    To find your subscription, follow the instructions in [Find your Azure subscription](https://learn.microsoft.com/en-us/azure/azure-portal/get-subscription-tenant-id#find-your-azure-subscription).

Make sure that default security group in your application Azure Virtual Network (VNet) allows internal connectivity. Otherwise, your application may not be able to reach the endpoint.

To use Private Link to connect your cluster to an Azure VNet that hosts your application, first create a private service endpoint (PSE) on your cluster, then create an endpoint in Azure.

## Create a PSE in YugabyteDB Managed

To create a PSE, do the following:

1. Enter the following command:

    ```sh
    ybm cluster network endpoint create \
      --cluster-name <yugabytedb_cluster> \
      --region <cluster_region> \
      --accessibility-type PRIVATE_SERVICE_ENDPOINT \
      --security-principals <azure_subscription_ids>
    ```

    Replace values as follows:

    - `yugabytedb_cluster` - name of your cluster.

>     - `cluster_region` - cluster region where you want to place the PSE. Must match one of the regions where your cluster is deployed (for example, `us-west-2`), and preferably match the region where your application is deployed.

Reviewer 2: Change "us-west-2" to "westus3".
